__tests__/components/HarvestStatisticsPanel.spec.ts

```diff
@@ -0,0 +1,299 @@
+    const mockHarvestingStore = {
+        namespaced: true,
+        state: {
+            status: defaultHarvestingStatus,
+            harvestedBlockStats: harvestedBlockStats,
+            isFetchingHarvestedBlockStats: false,
+            pollingTrials: 0,
+            currentSignerHarvestingModel: currentSignerHarvestingModel,
+        },
+        getters: {
+            status: (state) => state.status,
+            harvestedBlockStats: (state) => state.harvestedBlockStats,
+            isFetchingHarvestedBlockStats: (state) => state.isFetchingHarvestedBlockStats,
+            pollingTrials: (state) => state.pollingTrials,
+            currentSignerHarvestingModel: (state) => state.currentSignerHarvestingModel,
+        },
+        mutations: {
+            status: (state, status) => (state.status = status),
+            harvestedBlockStats: (state, harvestedBlockStats) => (state.harvestedBlockStats = harvestedBlockStats),
+        },
+    };
+    const networkCurrency = mosaicsMock[0];
+    const mockMosaicStore = {
+        namespaced: true,
+        state: {
+            networkCurrency: networkCurrency,
+        },
+        getters: {
+            networkCurrency: (state) => state.networkCurrency,
+        },
+    };
+
+    const renderComponent = (stateChanges?: { [field: string]: any }, dispatch?: () => any) => {
+        const store = getStore(
+            {
+                account: mockAccountStore,
+                harvesting: mockHarvestingStore,
+                mosaic: mockMosaicStore,
+            },
+            stateChanges,
+            dispatch,
+            false,
+            undefined,
+            true,
+        );
+
+        return {
+            ...render(HarvestStatisticsPanel, {
+                store,
+                i18n,
+            }),
+            store,
+        };
+    };
+
+    test('renders component', async () => {
+        // Arrange + Act:
+        const { getByText, getByTestId } = renderComponent();
+
+        // Assert:
+        expect(getByText(i18n.t('harvesting_status').toString())).toBeDefined();
+        expect(getByText(i18n.t('blocks_made').toString())).toBeDefined();
+        expect(getByText(i18n.t('fees_collected').toString())).toBeDefined();
+        expect(getByTestId('status-text').textContent).toBe(i18n.t('harvesting_status_inactive').toString());
+        expect(getByTestId('total-block-count').textContent.trim()).toBe('1000');
+        expect(getByTestId('total-fees-earned').textContent.trim()).toBe('0');
+    });
+
+    Object.keys(HarvestingStatus).map((key) => {
+        test(`harvesting status is changed to ${key} when refresh button is clicked`, async () => {
+            // Arrange:
+            const dispatch = jest.fn();
+            const { getByTestId, getByRole, store } = renderComponent({}, dispatch);
+
+            dispatch.mockImplementationOnce((type: string) => {
+                if (type === 'harvesting/FETCH_STATUS') {
```

Comment:
aren't you explicitly changing the status here?
it seems like you're hooking this event and then forwarding to store?
can we not use the "real" dispatcher here?

Reply:
we are mocking the vuex store in our testing where the vuex is involved. And this is part of it.
when the component dispatches `harvesting/FETCH_STATUS` event, it's expected to fetch(remote call) , calculate the new status and commit to the store.
since we want to mock this behaviour, after the first render before the refresh button is clicked we are setting the scene, so that the refresh event can set the correct status.

> can we not use the "real" dispatcher here?

I'm afraid not.